### PR TITLE
update description about dumpling privileges

### PR DESCRIPTION
**Pull request by shichun-0415**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
The previous description about privileges required for exporting data from Dumpling is not clear. This PR aims to detail the requirements in a more clear way.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/10884
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: dumpling-overview.md [hunk 1]
```diff
@@ -65,11 +65,10 @@ In the above commands, you need to modify `~/.bash_profile` to the path of your
 
 ### Required privileges
 
-- SELECT
-- RELOAD
-- LOCK TABLES
-- REPLICATION CLIENT
-- PROCESS
+- SELECT: Required when exporting tables.
+- RELOAD: Required when using `consistency flush`. RDS databases do not support this privilege. Therefore, you can ignore this privilege if the upstream is an RDS database or a managed service.
```
Comment (Frank945946) [inline dumpling-overview.md:69]: RDS （exclude TiDB ）

**Review by Frank945946 — APPROVED** [review]

File: dumpling-overview.md (hunk 1 above)
Comment (shichun-0415) [inline dumpling-overview.md:69]:
```suggestion
- RELOAD: Required when using `consistency flush`. Note that only TiDB supports this privilege. When the upstream is an RDS database or a managed service, you can ignore this privilege.
```

File: dumpling-overview.md [hunk 2]
```diff
@@ -65,11 +65,10 @@ In the above commands, you need to modify `~/.bash_profile` to the path of your
...
 
-- SELECT
-- RELOAD
-- LOCK TABLES
-- REPLICATION CLIENT
-- PROCESS
+- SELECT: Required when exporting tables.
+- RELOAD: Required when using `consistency flush`. Note that only TiDB supports this privilege. When the upstream is an RDS database or a managed service, you can ignore this privilege.
+- LOCK TABLES: Required when using `consistency lock`. This privilege must be granted for all the databases and tables to be exported.
+- REPLICATION CLIENT: Required when exporting metadata to record data snapshot. This privilege is optional.
```
Comment (ran-huang) [inline dumpling-overview.md:71]:
```suggestion
- REPLICATION CLIENT: Required when exporting metadata to record data snapshot. This privilege is optional if you do not need to export metadata.
```

**Review by ran-huang — APPROVED** [review]
Rest LGTM

File: dumpling-overview.md (hunk 2 above)
Comment (shichun-0415) [reply dumpling-overview.md:71]: IMO, optional is a fixed attribute and does not change if the user wants to export metadata. How about:
> This privilege is optional and you can ignore it if you do not need to export metadata.

Comment (ran-huang) [reply dumpling-overview.md:71]: LGTM

**Pull request merged**